app/models/mixins/supports_feature_mixin.rb

```diff
@@ -52,14 +54,60 @@ module SupportsFeatureMixin
   #
   extend ActiveSupport::Concern
 
+  QUERYABLE_FEATURES = {
+    :associate_floating_ip    => 'Associate a Floating IP',
+    :control                  => 'Basic control operations', # FIXME: this is just a internal helper and should be refactored
+    :delete                   => 'Deletion',
+    :disassociate_floating_ip => 'Disassociate a Floating IP',
+    :discovery                => 'Discovery of Managers for a Provider',
+    :evacuate                 => 'Evacuation',
+    :events                   => 'Query for events',
+    :launch_cockpit           => 'Launch Cockpit UI',
+    :live_migrate             => 'Live Migration',
+    :migrate                  => 'Migration',
+    :provisioning             => 'Provisioning',
+    :reconfigure              => 'Reconfiguration',
+    :regions                  => 'Regions of a Provider',
+    :resize                   => 'Resizing',
+    :retire                   => 'Retirement',
+    :smartstate_analysis      => 'Smartstate Analaysis',
+  }.freeze
+
+  # Whenever this mixin is included we define all features as unsupported by default.
+  # This way we can query for every feature
+  included do
+    QUERYABLE_FEATURES.keys.each do |feature|
+      supports_not(feature)
+    end
+  end
+
+  class UnknownFeatureError < StandardError; end
+
+  def self.guard_queryable_feature(feature)
+    unless QUERYABLE_FEATURES.key?(feature.to_s.to_sym)
+      raise UnknownFeatureError, "Feature ':#{feature}' is unknown to SupportsFeatureMixin."
```

Comment:
you are never too late :)

In this PR we dont handle the case where somebody just _asks_. But only if he really codes against something like `supports?(feature)`. So it's reasonable to raise an error, because he is coding against an unknown feature.

For _asking_ about known features I have this: https://github.com/ManageIQ/manageiq/pull/11179 - although it might be even debatable if we want such discoverability, because in the end the system should know what features it knows about